### ISSUE TITLE
scala-native support

### ISSUE THREAD

**Issue by xuwei-k** [monix/minitest#12]

**Comment by xuwei-k** [comment]
https://github.com/monix/minitest/pull/15

**Comment by alexandru** [comment]
@xuwei-k is there anything that I can do?

We currently depend on macro-compat for Scala 2.10, but Scala 2.10 is very obsolete at this point.

If we dropped Scala 2.10, would it be easy to support Scala Native?

**Comment by tindzk** [comment]
@alexandru Agreed. These days most projects are already using 2.11/2.12, so I would presume that 2.10 support can be safely dropped.

At the moment, Scala Native only supports Scala 2.11. You can have a look at [sbt-crossproject](https://github.com/portable-scala/sbt-crossproject) which makes it straighforward to support Scala Native.

**Comment by olafurpg** [comment]
@alexandru With the settings

```scala
.nativeSettings(
  scalaVersion := "2.11.12",
  crossScalaVersions := List("2.11.12")
)
```

you can cross-build the native modules to 2.11 only while other modules continue to cross-build for 2.10<->2.12.

**Comment by xuwei-k** [comment]
https://github.com/monix/minitest/pull/22

**Comment by ekrich** [comment]
Now that 2.12/2.13 support is available with `0.4.0` is this a reconsideration?
